### PR TITLE
subcmd/apply: Support fail-safe approach

### PR DESCRIPTION
**Pull request by shimon-armis**
## Description
Up until now, the `apply` command implemented a fail-fast strategy. This means that an attempt to configure a batch of topics (by passing a path to a folder contains a list of topics configurations files) will be terminated upon the first failure (w/o even trying to apply the configuration for the next-in-line topics).
This commit introduces the fail-safe support:
It allows the user to run the `apply` command in a fail-safe manner, which means that instead of failing upon the first failure, we'll simply continue to the next topic configuration, while aggregating all the errors along the way.
Eventually, if there were any errors, all of them will be shown.

### CODE REVIEW

**Comment by shimon-armis** [comment]
@hhahn-tw Can you please review this one? :)

**Comment by shimon-armis** [comment]
@hhahn-tw @petedannemann Can you guys maybe take a look?

**Comment by shimon-armis** [comment]
@jkoelker Can you maybe review this PR?

File: cmd/topicctl/subcmd/apply.go [hunk 1]
```diff
@@ -160,7 +177,10 @@ func applyRun(cmd *cobra.Command, args []string) error {
 		for _, match := range matches {
 			matchCount++
 			if err := applyTopic(ctx, match, adminClients); err != nil {
-				return err
+				if applyConfig.failFast {
+					return err
+				}
+				aggregatedError = appendError(aggregatedError, err)
```
Comment (jkoelker) [inline cmd/topicctl/subcmd/apply.go:183]:
```suggestion
				errs = errors.Join(errs, err)
```

File: cmd/topicctl/subcmd/apply.go [hunk 2]
```diff
@@ -169,7 +189,7 @@ func applyRun(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("No topic configs match the provided args (%+v)", args)
 	}
 
-	return nil
+	return aggregatedError
```
Comment (jkoelker) [inline cmd/topicctl/subcmd/apply.go:192]:
```suggestion
	return errs
```

**Review by jkoelker — APPROVED** [review]
LGTM, just a reccomendation to use `errors.Join` to concat the errors, if not, you can also use `fmt.Errorf` with multiple `%w` to achieve the same thing and not drop the error context by rendering it as a string.

File: cmd/topicctl/subcmd/apply.go (hunk 1 above)
Comment (shimon-armis) [reply cmd/topicctl/subcmd/apply.go]: Done.

File: cmd/topicctl/subcmd/apply.go [hunk 3]
```diff
@@ -138,6 +153,8 @@ func applyRun(cmd *cobra.Command, args []string) error {
 
 	// Keep a cache of the admin clients with the cluster config path as the key
 	adminClients := map[string]admin.Client{}
+	// Keep track of any errors that occur during the apply process
+	var aggregatedError error
```
[the remark this answers is not in the document]
Comment (shimon-armis) [reply cmd/topicctl/subcmd/apply.go]: Done.

File: cmd/topicctl/subcmd/apply.go (hunk 2 above)
Comment (shimon-armis) [reply cmd/topicctl/subcmd/apply.go]: Done.

**Comment by shimon-armis** [comment]
> LGTM, just a reccomendation to use `errors.Join` to concat the errors, if not, you can also use `fmt.Errorf` with multiple `%w` to achieve the same thing and not drop the error context by rendering it as a string.



> LGTM, just a reccomendation to use `errors.Join` to concat the errors, if not, you can also use `fmt.Errorf` with multiple `%w` to achieve the same thing and not drop the error context by rendering it as a string.



> LGTM, just a reccomendation to use `errors.Join` to concat the errors, if not, you can also use `fmt.Errorf` with multiple `%w` to achieve the same thing and not drop the error context by rendering it as a string.



> LGTM, just a reccomendation to use `errors.Join` to concat the errors, if not, you can also use `fmt.Errorf` with multiple `%w` to achieve the same thing and not drop the error context by rendering it as a string.

errors.Join() is only supported from go 1.20 (this repo is running a lower version).

**Comment by shimon-armis** [comment]
@hhahn-tw Can you please merge this one?

**Comment by hhahn-tw** [comment]
Thanks once more for your contribution!

**Pull request merged**